assets/javascript/game2.js

```diff
@@ -1,141 +1,69 @@
 $( document ).ready(function() {
     //console.log( "ready!" );
 
-//global variables
-  let wins = 0;
-  let losses = 0;
-  let numberToGuess;
-  let playerTotal = 0;
-  let value;
+    //global variables
+    const reset = init;
+    let wins = 0;
+    let losses = 0;
+    let numberToGuess;
+    let playerTotal;
 
-  
-function reset (){
-    $(".crystal").each(function(){
-        numberToGuess = Math.floor(Math.random() * 101 + 19);
-        //console.log(value);
-        {
-            $(this).attr({
-                "data-random":numberToGuess
-            });
-        }
-        //reset player score
-        playerTotal = 0;
-
-        //update total score
-        $("#totalScore").text(playerTotal);
-
-        //update random number to guess
-        $("#randomNumber").text(numberToGuess);
-    })
-    
-
-
-}
-reset()
+    init();
 
+    function generateNumberToGuess() {
+        return Math.floor(Math.random() * 101 + 19);
+    }
 
-//game start
-function gameStart(){
-    $(".crystal").click(function(){
-       value = Math.floor(Math.random() * 11 + 1);
-       $(this).attr({
-           
-       })
-       console.log(value);
-       playerTotal = playerTotal + value;
-       $("#totalScore").text(playerTotal);
-       winLoss();
-
+    function generateCrystalValue() {
+        return Math.floor(Math.random() * 11 + 1);
+    }
 
-    })
     
-}
-gameStart()
-
-
-
-function winLoss(){
-    if (playerTotal === numberToGuess){
-
-        //dsiplay win overlay
-        //winOn();
+    function init() {
+        // initialize playerTotal and numberToGuess
+        playerTotal = 0;
+        numberToGuess = generateNumberToGuess();
         
-        //update wins total
-        wins++;
-
-        //update wins display
-        $("#wins").text("Wins:" + wins);
-
-        //reset game
-        reset();
-
-    } else if (playerTotal > numberToGuess) {
-
-        //display lose overlay
-        //loseOn();
-
-        //update losses total
-        losses++;
+        // update DOM
+        $("#totalScore").text(playerTotal);
+        $("#randomNumber").text(numberToGuess);
 
-        //update losses display
-        $("#losses").text("Losses:" + losses);
+        $(".crystal").each((_index, crystal) => {
+            // declare crystal's value outside click handler, so it's the same value each time
+            const value = generateCrystalValue();
+        
+            // remove any existing click handlers
+            $(crystal).off("click");
 
-        //reset game
-        reset();
+            // add new click handler
+            $(crystal).on("click", (event) => {
+                event.preventDefault(); // this prevents the browser from "jumping" to the top of the page when a `crystal` is clicked
+                updateScore(value);
+            });
+        });
     }
 
-    }
-    winLoss()
```

Comment:
No reason to invoke `winLoss` here, since we know it's not going to do anything (i.e. `playerTotal`'s value is `0` when the thread of execution makes it here).